File: Alternating Factorial/AlternatingFactorial.py
print("This programs finds the nth term in the alternating sequence")
n=int(input("Enter the nth term whoose value you want to find"))

def find_fac(n):
    if(n==1):
        return(1)
    else:
        return(n*find_fac(n-1))

def alternating_fac(n):
    if(n==0 or n==1):
        return 1
    term=0
    flag_to_sub=0
    n=n+1
    while(n>0):
        if( flag_to_sub):
            term=term-(find_fac(n))
            flag_to_sub=0
        else:
            term=term+(find_fac(n))
            flag_to_sub=1

        n=n-1
        # print("Term",term)

    return term
        
            


print(alternating_fac(n))

File: Alternating Factorial/AlternatingFactorial.m
% Author : Paresh Yeole
% Matlab script to determine the nth term in the alternating 
% factorial sequence

disp("This programs finds the nth term in the alternating sequence");
n=input("\nEnter the nth term whoose value you want to find");
disp(alternating_number(n+1))

function result = alternating_number(n)
    if n==0 || n==1
        result = 1;
    else
        flag = 0;
        result = 0;
        while n>0
            if flag
                result = result - fact(n);
                flag = 0;
            else
                result = result + fact(n);
                flag = 1;
            end
            n = n - 1;
        end
    end
end


function fac = fact(n)
    if n==1 || n==0 
        fac = 1;
    else
        fac = n*fact(n-1); 
    end
end